djoser/serializers.py

```diff
@@ -145,6 +145,7 @@ def get_user(self, is_active=True):
         if (
             settings.PASSWORD_RESET_SHOW_EMAIL_NOT_FOUND
             or settings.USERNAME_RESET_SHOW_EMAIL_NOT_FOUND
+            or settings.RESEND_ACTIVATION_SHOW_EMAIL_NOT_FOUND
```

Comment:
As noted in #656, I think this logic is flawed. Setting any one of these settings to `True` has the same effect as setting them all to `True`. These settings imply a granularity that is not actually achieved through code.

It's debatable if these settings should even be there. Account Enumeration is widely regarded as a security flaw, and these flows possibly disclosing the existence of an account are an attack vector.

(note, I'm not a maintainer of Djoser, this is just my opinion, I defer any decision about this to the actual maintainers.)

Reply:
Duly noted. I am happy to make changes to address more granularity or remove these settings altogether. Not sure how the maintainers want to move forward though.

Reply:
> As noted in #656, I think this logic is flawed. Setting any one of these settings to True has the same effect as setting them all to True. These settings imply a granularity that is not actually achieved through code.

I agree but to me it sounds like a follow up. A small refactor is somewhere on the roadmap, I need to get familiar with the codebase again.

> It's debatable if these settings should even be there. Account Enumeration is widely regarded as a security flaw, and these flows possibly disclosing the existence of an account are an attack vector.

Public tools have different requirements than internal tools. Defaults are safe so I see no harm being done by just having them. 

@jaredleishman would you be willing to just drop this new setting? Your changes with views will mitigate 400 if no user exists and I agree that adding `RESEND_ACTIVATION_SHOW_EMAIL_NOT_FOUND` might not be the best way forward.

Reply:
@jaredleishman bump

Reply:
Hey @tomwojcik, sorry I didn't see your response to this thread until now. I've removed that new setting.

Reply:
No worries, thanks for pushing the requested changes. LGTM, just code quality step needs to be fixed.